tests/test_metadata.py

```diff
@@ -91,3 +93,49 @@ def test_write(tmpdir):
     data = json.loads(raw)
     assert data
     assert data["updateTime"] is not None
+
+
+def test_new_files(tmpdir):
+    metadata.reset()
+
+    # Create some files in nested directories:
+    # old: src/a
+    # new: src/code/b
+    # new: src/code/c
+    srcdir = tmpdir / "src"
+    os.mkdir(srcdir)
+    codedir = srcdir / "code"
+    os.mkdir(codedir)
+    with open(srcdir / "a", "wt") as file:
+        file.write("a")
+    time.sleep(1)
+    after_a_before_b = time.time()
+    time.sleep(1)
+    b_path = os.path.join(codedir, "b")
+    with open(b_path, "wt") as file:
+        file.write("b")
+    time.sleep(1)
+    after_b_before_c = time.time()
+    time.sleep(1)
+    c_path = os.path.join(codedir, "c")
+    with open(c_path, "wt") as file:
+        file.write("c")
+
+    # Confirm add_new_files found the new files and ignored the old one.
```

Comment:
I want be tempted to break this out into a couple assertions, perhaps you could create a `create_test_directory` helper, and then have a:

`test_add_new_files`, and `remove_obsolete_files` assertion?

Reply:
Done.